assets/_js/custom-viz/boeing/waypoints.js

```diff
@@ -85,25 +89,50 @@ const Waypoints = () => {
     })
   })
 
-  //if paragraph hits top of screen, part label disappears
   paragraphs.forEach(paragraph => {
-    const allParts = Array.from(document.querySelectorAll('.part'))
+    // const allParts = Array.from(document.querySelectorAll('.part'))
     const activeParts = Array.from(paragraph.querySelectorAll('.part'))
+    // const activeSVGs = []
 
     new Waypoint({
       element: paragraph,
       handler: function() {
-        allParts.forEach(function(part) {
+        const previousWaypoint = this.previous()
+
+        activeParts.forEach(function(part) {
           let partName = part.getAttribute('data-part')
           let partLabel = document.getElementById(`${partName}-label`)
+          let partSVG = document.getElementById(`${partName}-part`)
 
-          if (activeParts.indexOf(part) < 0) {
-            partLabel.classList.add('js-hidden')
-            console.log(part + ' are the activeParts')
-          } else {
-            partLabel.classList.remove('js-hidden')
+          partLabel.classList.remove('js-hidden')
+
+          if (
+            partSVG !== null &&
+            `${partName}-part` === partSVG.getAttribute('id')
+          ) {
+            partSVG.classList.remove('boeing-js__opacity-not-viewed')
+            partSVG.classList.add('boeing-js__opacity-focused')
           }
         })
+
+        if (previousWaypoint) {
+          const previousNodes = previousWaypoint.element.children
+          Array.from(previousNodes).forEach(child => {
+            if (child.classList.contains('part')) {
+              document
+                .getElementById(`${child.getAttribute('data-part')}-label`)
+                .classList.add('js-hidden')
+
+              let previousChild = document.getElementById(
+                `${child.getAttribute('data-part')}-part`
+              )
+              if (previousChild !== null) {
```

Comment:
Functionally, this is all fine. However, you might want to consider refactoring this. Generally speaking, having 3 nested `if` statements isn't ideal. Rather, you should look and see where you could do early returns. For example,

```js
if (!previousWaypoint) {
  return
}

// Rest of code here, although it also could be broken up into early returns.
```